src/main/java/be/seeseemelk/mockbukkit/MockBukkitExtension.java

```diff
@@ -0,0 +1,136 @@
+package be.seeseemelk.mockbukkit;
+
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.jetbrains.annotations.ApiStatus.Experimental;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Parameter;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Extension that mocks the Bukkit singleton before each test and subsequently unmocks it after each test. It will also
+ * inject this instance of {@link ServerMock} to any field or parameter of that type in the extended test class.
+ *
+ * <p>Example field usage:</p>
+ *
+ * <pre class="code"><code class="java">
+ * <b>&#064;ExtendWith(MockBukkitExtension.class)</b>
+ * class FieldExampleTest
+ * {
+ *
+ * 	private ServerMock serverMock;
+ *
+ * 	&#064;Test
+ * 	void aUnitTest()
+ *	{
+ *		assert serverMock != null;
+ *		// ...
+ *	}
+ *
+ * }
+ * </code></pre>
+ * <p>
+ * Example constructor parameter usage:
+ *
+ * <pre class="code"><code class="java">
+ * <b>&#064;ExtendWith(MockBukkitExtension.class)</b>
+ * class ConstructorExampleTest
+ * {
+ *
+ * 	private ServerMock serverMock;
+ *
+ * 	public ConstructorExampleTest(ServerMock serverMock)
+ *	{
+ * 		this.serverMock = serverMock;
+ *	}
+ *
+ * 	&#064;Test
+ * 	void aUnitTest()
+ *	{
+ * 		assert serverMock != null;
+ * 		// ...
+ *	}
+ *
+ * }
+ * </code></pre>
+ * <p>
+ * Example method parameter usage:
+ *
+ * <pre class="code"><code class="java">
+ * <b>&#064;ExtendWith(MockBukkitExtension.class)</b>
+ * class MethodExampleTest
+ * {
+ *
+ * 	&#064;Test
+ * 	void aUnitTest(ServerMock serverMock)
+ *	{
+ * 		assert serverMock != null;
+ * 		// ...
+ *	}
+ *
+ * }
+ * </code></pre>
+ */
+@Experimental
+public class MockBukkitExtension implements BeforeEachCallback, AfterEachCallback, ParameterResolver
+{
+
+	@Override
+	public void beforeEach(ExtensionContext context) throws Exception
+	{
+		MockBukkit.getOrCreateMock();
+		injectServerMockIntoFields(context);
+	}
+
+	private void injectServerMockIntoFields(ExtensionContext context) throws IllegalAccessException
+	{
+		final Optional<Class<?>> classOptional = context.getTestClass();
+		if (classOptional.isEmpty())
+			return;
+
+		final List<Field> serverMockFields = FieldUtils.getAllFieldsList(classOptional.get())
+				.stream()
+				.filter(field -> field.getType() == ServerMock.class)
```

Comment:
It may be a wise idea to also require an annotation for fields, as it may lead to unexpected results and confusion if devs do not explicitly define this behaviour.

Reply:
I've added `MockBukkitServer` back but I'm still not fully convinced that it's the right name. What do you think?

Another option could be something like `MockBukkitInject`, which could open the door to more injected objects in the future (but could create more confusion in the present since only `ServerMock` is supported).

Reply:
Hey apologies to bug you but any thoughts on the above?

Reply:
I like the idea of calling it `MockBukkitInject`. I don't think it would cause much confusion, and I do like the idea of being able to inject more objects in the future (perhaps Worlds, Players, other entities, ...).

Reply:
Ok I'll push a commit to rename it!